proto_text.c

```diff
@@ -1676,6 +1676,7 @@ static void process_mdelete_command(conn *c, token_t *tokens, const size_t ntoke
     }
 cleanup:
     if (it) {
+        LOGGER_LOG(NULL, LOG_DELETIONS, LOGGER_DELETIONS, it, META_DELETE);
```

Comment:
this will log from a delete miss, which I think your PR says this shouldn't do?

Reply:
oh I did test a miss scenario and it didn't log. I will test it again

Reply:
tried `md` on a `NF` response and it didn't log it

Reply:
commands: 
```log
Trying 127.0.0.1...
Connected to localhost.
Escape character is '^]'.
delete 121
NOT_FOUND
delete adslkjflskjdf
NOT_FOUND
md aljdfklsf
NF
ms 1231 1
1
HD
md 1231
HD
```
corresponding deletions:
```
echo watch deletions | nc localhost 11211
OK
ts=1678391997.828498 gid=2 type=deleted key=1231 cmd=md clsid=1 size=1
```

Reply:
looking from code block at high level it seems:
```c
it = item_get_locked(...)
if (it) {
   // check if to mark stale or actually delete
   stats.delete_hit++;
   goto cleanup;
} else {
   stats.delete_misses++;
   goto cleanup;
}

cleanup:
if (it) {
   LOGGER.log();
   do_item_remove();
}
```

can you confirm?

Reply:
not having the `it` seems to cause a delete miss and the log is within the condition.

Reply:
oh - yes that's valid point!

Reply:
updated - also added a test to test this behavior